src/java.compiler/share/classes/javax/lang/model/util/package-info.java

```diff
@@ -35,6 +35,274 @@
+ *  +    }
+ *
+ * //  Sample diff for SimpleElementVisitorPreview.java
+ * //  Take the default action for a preview feature 1.
+ *  +    /**
+ *  +     * {&commat;inheritDoc ElementVisitor}
+ *  +     *
+ *  +     * &commat;implSpec Visits a {&commat;code PreviewFeature1Element} by calling
+ *  +     * {@code defaultAction}.
+ *  +     *
+ *  +     * &commat;param e {&commat;inheritDoc ElementVisitor}
+ *  +     * &commat;param p {&commat;inheritDoc ElementVisitor}
+ *  +     * &commat;return  {&commat;inheritDoc ElementVisitor}
+ *  +     * &commat;since N
+ *  +     *&sol;
+ *  +    @Override
+ *  +    public R visitPreviewFeature1(PreviewFeature1Element e, P p) {
+ *  +        return defaultAction(e, p);
+ *  +    }
+ * </pre>
+ *
+ * When preview feature 1 exits preview in JDK (<i>N+k</i>), a set of
+ * visitors for language level (<i>N+k</i>) would be added. The
+ * methods operating over the feature would be moved from the preview
+ * visitors to the new language level (<i>N+k</i>) visitors. Each
+ * preview visitor would then have its direct superclass changed to
+ * the new corresponding (<i>N+k</i>) visitor.
+ *
+ * <h3 id=newKindLangConstruct>Adding visitor support for a language
+ * construct that is a new kind of an existing construct</h3>
+ *
+ * Consider a new language feature, preview feature 2, in JDK
+ * <i>N</i>. This feature has a new element kind <em>without</em> a
+ * new top-level element interface needed to model it. Concretely,
+ * assume a preview feature 2 is a new kind of variable; the changes
+ * would be analogous if the feature were a new kind of executable
+ * instead or new kind of another existing top-level construct. In
+ * that case, the API changes are more limited:
+ *
+ * <pre>
+ *  //  Sample diff for ElementKind.java
+ *  +    /**
+ *  +     * A preview feature 2.
+ *  +     * &commat;since N
+ *  +     *&sol;
+ *  +     PREVIEW_FEATURE_2,
+ *  ...
+ *  // Update existing methods as needed
+ *       public boolean isVariable() {
+ *           return switch(this) {
+ *           case ENUM_CONSTANT, FIELD, PARAMETER,
+ *                LOCAL_VARIABLE, EXCEPTION_PARAMETER, RESOURCE_VARIABLE,
+ *  -             BINDING_VARIABLE -> true;
+ *  +            BINDING_VARIABLE, PREVIEW_FEATURE_2 -> true;
+ *           default -> false;
+ *           };
+ *       }
+ * </pre>
+ *
+ * The kind visitors need support for the new variety of element:
+ * <pre>
+ * // Update visitVariable in ElementKindVisitor6:
+ *        ...
+ *        * &commat;implSpec This implementation dispatches to the visit method for
+ *        * the specific {&commat;linkplain ElementKind kind} of variable, {&commat;code
+ *        * ENUM_CONSTANT}, {&commat;code EXCEPTION_PARAMETER}, {&commat;code FIELD},
+ *  -     * {&commat;code LOCAL_VARIABLE}, {&commat;code PARAMETER}, or {&commat;code RESOURCE_VARIABLE}.
+ *  +     * {&commat;code LOCAL_VARIABLE}, {&commat;code PARAMETER}, or {&commat;code RESOURCE_VARIABLE},
```

Comment:
delete `or` at this position in the list

Reply:
Fixed; thanks.